src/main.ts

```diff
@@ -56,19 +56,24 @@ export async function run() {
       core.info(`Successfully set up Go version ${versionSpec}`);
     }
 
+    let goPath = await io.which('go');
+    let goVersion = (cp.execSync(`${goPath} version`) || '').toString();
+
```

Comment:
```suggestion
    const goVersion = await getInstalledGoVersion();
```
**Nit:** I suggest putting these two lines of code into the function parseGoVersion() and renaming this function to getInstalledGoVersion(). On line 59 we can just call getInstalledGoVersion() and then reuse its result down the code in restoreCache() (line 65) and in core.info() (line 77).